### PR TITLE
Compute normals on command line

### PR DESCRIPTION
**Pull request by TomCrowder**
This feature has been requested several times (http://www.cloudcompare.org/forum/viewtopic.php?f=9&t=2403) and I need it too!

I'm hoping it can be included in a release soon!

### CODE REVIEW

File: qCC/ccCommandLineCommands.h [hunk 1]
```diff
@@ -448,6 +450,60 @@ struct CommandLoad : public ccCommandLineInterface::Command
...
+		{
+			return cmd.error(QString("Missing radius parameter"));
+		}
+
+		cmd.print(QString("\tRadius: %1").arg(radius));
+
+		CC_LOCAL_MODEL_TYPES model = QUADRIC;
+		ccNormalVectors::Orientation  orientation = ccNormalVectors::Orientation::UNDEFINED;
+		PointCoordinateType defaultRadius = radius;
+		for (size_t i = 0; i < cmd.clouds().size(); ++i)
```
Comment (asmaloney) [inline qCC/ccCommandLineCommands.h]: Can you use a range-for here? It would be cleaner.

File: qCC/ccCommandLineCommands.h [hunk 2]
```diff
@@ -448,6 +450,60 @@ struct CommandLoad : public ccCommandLineInterface::Command
...
+		{
+			ccPointCloud* cloud = cmd.clouds()[i].pc;
+			cmd.print("computeNormalsWithOctree started...\n");
+			bool success = cloud->computeNormalsWithOctree(QUADRIC, orientation, defaultRadius, nullptr);
+			if(success){
```
Comment (asmaloney) [inline qCC/ccCommandLineCommands.h]:
```
if (success )
{
}
else
{
}
```

File: qCC/ccCommandLineCommands.h [hunk 3]
```diff
@@ -448,6 +450,60 @@ struct CommandLoad : public ccCommandLineInterface::Command
...
 };
 
+struct CommandOctreeNormal : public ccCommandLineInterface::Command
+{
+	CommandOctreeNormal() : ccCommandLineInterface::Command("OCTREE_NORMALS", COMMAND_COMPUTE_OCTREE_NORMALS) {}
+
+	virtual bool process(ccCommandLineInterface& cmd) override
+	{
+		cmd.print("[OCTREE NORMALS CALCULATION]");
+		ccLog::Error("OCTREE NORMALS CALCULATION");
```
Comment (asmaloney) [inline qCC/ccCommandLineCommands.h]: Why is this logging an error?

**Comment by TomCrowder** [comment]
Thanks for the comments @asmaloney , I've made changes off the back of them

File: qCC/ccCommandLineCommands.h (hunk 2 above)
Comment (asmaloney) [inline qCC/ccCommandLineCommands.h]: This is a formatting issue. While you're fixing this, maybe fix indentation on the "for"?

**Comment by TomCrowder** [comment]
Great, formatting changes made; thanks!

**Comment by asmaloney** [comment]
Thanks Tom!

**Comment by dgirardeau** [comment]
I would put '<>' instead of quotes for #include "CCConst.h" and #include "ccLog.h" (as their are not in the same folder). But I'll do it myself afterwards, it will be simpler.

Thanks for the contribution!

**Pull request merged**